src/main/java/org/elasticsearch/kafka/indexer/jobs/ConsumerWorker.java

```diff
@@ -20,77 +22,86 @@
 import org.apache.kafka.clients.consumer.RetriableCommitFailedException;
 import org.apache.kafka.common.TopicPartition;
 import org.apache.kafka.common.errors.WakeupException;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.elasticsearch.kafka.indexer.CommonKafkaUtils;
 import org.elasticsearch.kafka.indexer.FailedEventsLogger;
 import org.elasticsearch.kafka.indexer.service.IBatchMessageProcessor;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.ObjectFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 
-public class ConsumerWorker implements Runnable, AutoCloseable {
+public class ConsumerWorker implements AutoCloseable, IConsumerWorker {
 
     private static final Logger logger = LoggerFactory.getLogger(ConsumerWorker.class);
-    private IBatchMessageProcessor batchMessageProcessor;
-    private final KafkaConsumer<String, String> consumer;
-    private final String kafkaTopic;
-    private final int consumerId;
-    // interval in MS to poll Kafka brokers for messages, in case there were no
-    // messages during the previous interval
+    @Value("${kafka.consumer.source.topic:testTopic}")
+    private String kafkaTopic;
+    @Value("${application.id:app1}")
+    private String consumerInstanceName;
+    // interval in MS to poll Kafka brokers for messages, in case there were no messages during the previous interval
+    @Value("${kafka.consumer.poll.interval.ms:10000}")
     private long pollIntervalMs;
+    @Value("${kafka.consumer.property.prefix:consumer.kafka.property.}")
+    private String consumerKafkaPropertyPrefix;
+    @Resource(name = "applicationProperties")
+    private Properties applicationProperties;
+
     private OffsetLoggingCallbackImpl offsetLoggingCallback;
+    private IBatchMessageProcessor batchMessageProcessor;
+    
+    private KafkaConsumer<String, String> consumer;
     private AtomicBoolean running = new AtomicBoolean(false);
-    private String consumerGroupName;
-
-    public ConsumerWorker(
-            int consumerId, 
-            String consumerInstanceName, 
-            String kafkaTopic, 
-            Properties kafkaProperties,
-            long pollIntervalMs, 
-            IBatchMessageProcessor batchMessageProcessor, 
-            OffsetLoggingCallbackImpl offsetLoggingCallback) {
-        this.batchMessageProcessor = batchMessageProcessor;
-        this.offsetLoggingCallback = offsetLoggingCallback;
-        this.consumerId = consumerId;
-        this.kafkaTopic = kafkaTopic;
-        this.pollIntervalMs = pollIntervalMs;
-        kafkaProperties.put(ConsumerConfig.CLIENT_ID_CONFIG, consumerInstanceName + "-" + consumerId);
+    private int consumerInstanceNumber;
+
+    public ConsumerWorker() {        
+    }
+    
+    @Override
+    public void initConsumerInstance(int consumerInstanceNumber) {
```

Comment:
it's rather consumer id, my first thought was that it's a number of consumers to create. little confusing

Reply:
done - renamed